packages/block-editor/src/hooks/auto-inserting-blocks.js

```diff
@@ -0,0 +1,214 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { addFilter } from '@wordpress/hooks';
+import { Fragment } from '@wordpress/element';
+import { PanelBody, ToggleControl } from '@wordpress/components';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { createBlock, store as blocksStore } from '@wordpress/blocks';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { InspectorControls } from '../components';
+import { store as blockEditorStore } from '../store';
+
+function AutoInsertingBlocksControl( props ) {
+	const autoInsertedBlocksForCurrentBlock = useSelect(
+		( select ) => {
+			const { getBlockTypes } = select( blocksStore );
+			return getBlockTypes()?.filter(
+				( block ) =>
+					block.autoInsert &&
+					Object.keys( block.autoInsert ).includes( props.blockName )
+			);
+		},
+		[ props.blockName ]
+	);
+
+	const {
+		autoInsertedBlockClientIds,
+		blockIndex,
+		rootClientId,
+		innerBlocksLength,
+	} = useSelect(
+		( select ) => {
+			const { getBlock, getBlockIndex, getBlockRootClientId } =
+				select( blockEditorStore );
+			const _rootClientId = getBlockRootClientId( props.clientId );
+
+			const _autoInsertedBlockClientIds =
+				autoInsertedBlocksForCurrentBlock.reduce( ( acc, block ) => {
+					const relativePosition =
+						block?.autoInsert?.[ props.blockName ];
```

Comment:
if we validate this here we could carry a type through and ensure we never check it to be something unsupported. we could import a type from a `types.ts` file

```ts
type AutoInsertionRelativePosition
  = 'before'
  | 'after'
  | 'first_child'
  | 'last_child'
  as const.
```

and then create a function to return that

```js
/**
 * Parses a relative position set in a block setting.
 *
 * @param {unknown} position Potential position read from external source.
 * @return {import('../../types.ts').AutoInsertionRelativePosition|null} A relative position if parsed, or `null` if invalid.
 */
function parseRelativePosition = ( position ) =>
	('before' === position || 'after' === position || 'first_child' === position || 'last_child' === position)
		? position
		: null;
```

I admit, that seems like a lot of junk just to add a type, but I think later on it could be an asset that lets us avoid making sure that everywhere we look we are checking the right values.

```js
const relativePosition = parseRelativePosition( block?.autoInsert?.[ props.blockName ] );
if ( null === relativePosition ) {
	return null;
}
```

otherwise it makes me kind of nervous seeing our `[ 'before', 'after' ].includes( … )` checks.

Reply:
I wish this kind of typing would read a bit less redundant in JS 😬 (where we still have to basically compare strings -- it feels like the new type doesn't add that much value in terms of how we can express this check 😢 ) 

What I've done for now to avoid the overhead is rewrite the remaining `[ 'before', 'after' ].includes( … )` check to a `switch` statement: 64c70af3c6. Think that'll do? 😊